masu/theme.json

```diff
@@ -0,0 +1,699 @@
+				},
+				{
+					"fluid": {
+						"max": "3.125rem",
+						"min": "2.1875rem"
+					},
+					"name": "2X Large",
+					"size": "2.65625rem",
+					"slug": "xx-large"
+				},
+				{
+					"fluid": {
+						"max": "3.75rem",
+						"min": "2.5625rem"
+					},
+					"name": "3X Large",
+					"size": "3.15625rem",
+					"slug": "xxx-large"
+				},
+				{
+					"fluid": {
+						"max": "4.5rem",
+						"min": "3.125rem"
+					},
+					"name": "4X Large",
+					"size": "3.8125rem",
+					"slug": "xxxx-large"
+				},
+				{
+					"fluid": {
+						"max": "5.375rem",
+						"min": "3.75rem"
+					},
+					"name": "5X Large",
+					"size": "4.5625rem",
+					"slug": "xxxxx-large"
+				}
+			]
+		}
+	},
+	"styles": {
+		"blocks": {
+			"core/button": {
+				"border": {
+					"color": "var(--wp--preset--color--primary)",
+					"radius": "0",
+					"style": "solid",
+					"width": "1px"
+				},
+				"color": {
+					"background": "var(--wp--preset--color--primary)",
+					"text": "var(--wp--preset--color--background)"
+				},
+				"spacing": {
+					"padding": "0.9375rem 1.9375rem"
+				},
+				"typography": {
+					"fontSize": "var(--wp--preset--font-size--x-small)",
+					"fontWeight": "normal",
+					"lineHeight": "var(--wp--custom--typography--line-height--tiny)"
+				}
+			},
+			"core/code": {
+				"border": {
+					"color": "var(--wp--preset--color--foreground)",
+					"radius": "0px",
+					"style": "solid",
+					"width": "1px"
+				},
+				"color": {
+					"background": "var(--wp--preset--color--foregound)"
+				},
+				"spacing": {
+					"padding": "var(--wp--custom--gap--vertical)"
+				},
+				"typography": {
+					"fontFamily": "monospace"
+				}
+			},
+			"core/columns": {
+				"spacing": {
+					"blockGap": "var(--wp--custom--gap--horizontal)"
+				}
+			},
+			"core/gallery": {
+				"spacing": {
+					"margin": {
+						"bottom": "var(--wp--custom--gap--vertical)"
+					}
+				}
+			},
+			"core/heading": {
+				"typography": {
+					"fontFamily": "var(--wp--preset--font-family--libre-franklin)",
+					"fontWeight": "700"
+				}
+			},
+			"core/list": {
+				"spacing": {
+					"padding": {
+						"left": "var(--wp--custom--gap--vertical)"
+					}
+				}
+			},
+			"core/navigation": {
+				"color": {
+					"text": "var(--wp--preset--color--foreground)"
+				},
+				"typography": {
+					"fontSize": "var(--wp--preset--font-size--x-small)",
+					"lineHeight": "var(--wp--custom--typography--line-height--normal)"
+				},
+				"elements": {
+					"link": {
+						"color": {
+							"text": "var(--wp--preset--color--primary)"
+						},
+						"typography": {
+							"textDecoration": "none"
+						},
+						":hover": {
+							"typography": {
+								"textDecoration": "underline dotted"
+							}
+						},
+						":active": {
+							"typography": {
+								"textDecoration": "underline solid"
+							}
+						}
+					}
+				}
+			},
+			"core/paragraph": {
+				"typography": {
+					"lineHeight": "calc( 1em + 0.875rem )"
```

Comment:
Hey there @iamtakashi 👋  I wonder if there is a reason why this doesn't have a number value or a `var` name and it just includes a `calc()` function? Currently, we are updating the mobile editor code to take into account these cases, but until that code goes live and the adoption of that future update reaches a nice percentage, this currently breaks for users who use this theme. Is there a chance we could update it to not have a `calc` function and just a value/var with a value instead?

Reply:
Hi @geriux 👋 Thanks for the ping. The reason why I used that was to make sure paragraphs look good regardless of the size. When we use a single value, it gets hideous and separated too much as the size increases.

| Current | static value (e.g.1.5) |
| ----------- | ----------- |
| ![nice](https://user-images.githubusercontent.com/908665/211880522-bd337283-12c0-4c44-8df6-f9adf0812d09.jpg) | ![ugly](https://user-images.githubusercontent.com/908665/211880569-a47289e4-e066-4eee-8e04-8f53b34a63b2.jpg) |

If I'm not wrong, the paragraph block doesn't allow specifying different line height values for different sizes, and I use this solution in other themes coming up.

How about like this? Would this work with the mobile editor code, or would this be the same? Do you have any suggestions?

```
"settings": {
	"custom": {
		"typography": {
			"line-height": {
				"test": "calc( 1em + 0.875rem )"
			}
		}
	}
}
"styles": {
	"blocks": {
		"core/paragraph": {
			"typography": {
				"lineHeight": "var(--wp--custom--typography--line-height--test)"
			}
		}
	}
}
```

Reply:
Hey there! Thanks for the quick reply and for also suggesting a workaround. After some investigation, we found we also had issues while parsing decimals in math expressions like in this one `calc( 1em + 0.875rem )` unfortunately this will need to be a fix for the app itself and not this actual theme because it could be a common use case in other themes configuration.

Let's keep it as it is now and we will support it correctly in an upcoming app release.

Thanks again!

Reply:
Got it. No worries!